mobile-app/app/screens/AppNavigator/screens/Balances/hooks/TokenPrice.ts

```diff
@@ -56,78 +35,10 @@ export function useTokenPrice (): TokenPrice {
-    if (result !== undefined) {
-      return new BigNumber(result).multipliedBy(amount)
-    }
-    // active price for walletTokens based on USDT
-    const arbitraryPoolPair = getArbitraryPoolPair(symbol, 'USDT')
-
-    if (arbitraryPoolPair.length > 0) {
-      const {
-        aToBPrice,
-        estimated
-      } = calculatePriceRates(symbol, arbitraryPoolPair, amount)
-      // store price for each unit in cache
-      CacheApi.set(key, aToBPrice.toFixed(8))
-      return estimated
-    }
-    return new BigNumber('')
-  }, [pairs, blockCount])
-
-  const getArbitraryPoolPair = useCallback((tokenASymbol: string, tokenBSymbol: string): PoolPairData[] => {
-    // TODO - Handle cheapest path with N hops, currently this logic finds the shortest path
-    const { path } = findPath(graph, tokenASymbol, tokenBSymbol)
-    return path.reduce((poolPairs: PoolPairData[], token, index): PoolPairData[] => {
-      const pair = pairs.find(pair => checkIfPair({
-        a: pair.data.tokenA.symbol,
-        b: pair.data.tokenB.symbol
-      }, token, path[index + 1]))
-      if ((pair == null) || index === path.length) {
-        return poolPairs
-      }
-      return [...poolPairs, pair.data]
-    }, [])
-  }, [pairs, blockCount])
-
-  const calculatePriceRates = useCallback((fromTokenSymbol: string, pairs: PoolPairData[], amount: BigNumber): CalculatePriceRatesI => {
-    let lastTokenBySymbol = fromTokenSymbol
-    let lastAmount = new BigNumber(amount)
-    const priceRates = pairs.reduce((priceRates, pair): { aToBPrice: BigNumber, bToAPrice: BigNumber, estimated: BigNumber } => {
-      const [reserveA, reserveB] = pair.tokenB.symbol === lastTokenBySymbol ? [pair.tokenB.reserve, pair.tokenA.reserve] : [pair.tokenA.reserve, pair.tokenB.reserve]
-      const [tokenASymbol, tokenBSymbol] = pair.tokenB.symbol === lastTokenBySymbol ? [pair.tokenB.symbol, pair.tokenA.symbol] : [pair.tokenA.symbol, pair.tokenB.symbol]
-
-      const priceRateA = new BigNumber(reserveB).div(reserveA)
-      const priceRateB = new BigNumber(reserveA).div(reserveB)
-      // To sequentially convert the token from its last token
-      const aToBPrice = tokenASymbol === lastTokenBySymbol ? priceRateA : priceRateB
-      const bToAPrice = tokenASymbol === lastTokenBySymbol ? priceRateB : priceRateA
-      const estimated = new BigNumber(lastAmount).times(aToBPrice)
-
-      lastAmount = estimated
-      lastTokenBySymbol = tokenBSymbol
-      return {
-        aToBPrice: priceRates.aToBPrice.times(aToBPrice),
-        bToAPrice: priceRates.bToAPrice.times(bToAPrice),
-        estimated
-      }
-    }, {
-      aToBPrice: new BigNumber(1),
-      bToAPrice: new BigNumber(1),
-      estimated: new BigNumber(0)
-    })
-
-    return {
-      aToBPrice: priceRates.aToBPrice,
-      bToAPrice: priceRates.bToAPrice,
-      estimated: priceRates.estimated
-    }
-  }, [pairs, blockCount])
+    return new BigNumber(prices[symbol]?.denominationPrice).multipliedBy(amount)
```

Comment:
do we need to have a fallback value if `prices[symbol]` is undefined?

Reply:
It will give `new BigNumber(undefined).toFixed(8)` as `NaN` and if NaN value is there then we are not showing its usd Breakdown, So i think the current logic will works fine 
cc @thedoublejay

Reply:
@fullstackninja864 Let's add a fallback to zero. just as a safety precaution